gnc/sub8_perception/src/sub8_pcl/cv_utils.cc

```diff
@@ -1,12 +1,270 @@
+	// std::cout << "high_thresh_search_start: " << target_mode << std::endl;
+	for(int i = 0; i < derivative_minima.size(); i++){
+		// std::cout << boost::format("derivative_minima[%1%].x = %2%  target_mode = %3%") % i % derivative_minima[i].x %  target_mode << std::endl;
+		if(derivative_minima[i].x > target_mode){
+			high_thresh_search_start = derivative_minima[i].x;
+			// std::cout << high_thresh_search_start << " Accepted" << std::endl;
+			break;
+		} 
+	}
+	// if(high_thresh_search_start == target_mode) cv::waitKey(0);
+	// std::cout << "low_thresh_search_start: " << target_mode << std::endl;
+	for(int i = derivative_maxima.size() - 1; i >= 0; i--){
+		// std::cout << boost::format("derivative_maxima[%1%].x = %2%  target_mode = %3%") % i % derivative_maxima[i].x %  target_mode << std::endl;
+		if(derivative_maxima[i].x < target_mode){
+			low_thresh_search_start = derivative_maxima[i].x;
+			// std::cout << low_thresh_search_start << " Accepted" << std::endl;
+			break;
+		}
+	}
+	// if(low_thresh_search_start == target_mode) cv::waitKey(0);
+	int high_thresh = high_thresh_search_start; int low_thresh = low_thresh_search_start;
+	// std::cout << std::endl << "high_deriv_thresh: " << hist_deriv_stddev[0] << " * " << high_thresh_gain << " = " << high_abs_derivative_thresh << std::endl;
+	// std::cout << "abs(high_deriv_thresh) - abs(slope) = slope_error" << std::endl;
+	for(int i = high_thresh_search_start; i < hist_size; i++){
+		int abs_slope = std::abs(hist_derivative.at<float>(i));
+		// std::cout << "i = " << i << "  :  " << high_abs_derivative_thresh << " - " << abs_slope << " = " << high_abs_derivative_thresh - abs_slope << std::endl;
+		if(abs_slope < high_abs_derivative_thresh){
+			high_thresh = i;
+			break;
+		}
+	}
+	// std::cout << "high_thresh = " << high_thresh << std::endl;
+	// std::cout << std::endl << "low_deriv_thresh: " << hist_deriv_stddev[0] << " * " << low_thresh_gain << " = " << low_abs_derivative_thresh << std::endl;
+	// std::cout << "abs(low_deriv_thresh) - abs(slope) = slope_error" << std::endl;
+	for(int i = low_thresh_search_start; i > 0; i--){
+		int abs_slope = std::abs(hist_derivative.at<float>(i));
+		// std::cout << "i = " << i << "  :  " << low_abs_derivative_thresh << " - " << abs_slope << " = " << high_abs_derivative_thresh - abs_slope << std::endl;
+		if(abs_slope < low_abs_derivative_thresh){ 
+			low_thresh = i;
+			break; 
+		}
+	}
+	// std::cout << "low_thresh = " << low_thresh << std::endl;
+	// std::cout << std::endl;
+	std::string ros_log = 
+		( boost::format("Target: %1%\nClosest distribution mode: %2%  Thresholds selected:  low=%3%  high=%4%")
+		% target % target_mode % low_thresh % high_thresh ).str() ;
+	ROS_INFO(ros_log.c_str());
+
+	// Threshold image
+	cv::inRange(src, low_thresh, high_thresh, dest);
+
+	#ifdef VISUALIZE
+	// Prepare to draw graph of histogram and derivative
+	int hist_w = 512; int hist_h = 400;
+	int bin_w = cvRound( (double) hist_w/hist_size );
```

Comment:
Counterexample: 50.0/26.0 = 1.9231 which gets rounded up to 2.
50 / 26 = 1 (integer division)